apps/allocations/app/components/App/App.js

```diff
@@ -108,26 +165,71 @@ const App = () => {
     children: PropTypes.node.isRequired,
   }
 
-  if (accounts.length === 0) {
+  if (budgets.length === 0) {
     return <Wrap><Empty action={onNewBudget} /></Wrap>
   }
 
   return (
-    // TODO: Profile App with React.StrictMode, perf and why-did-you-update, apply memoization
     <Wrap>
       <Header
         primary="Allocations"
         secondary={
           <Button mode="strong" icon={<IconPlus />} onClick={onNewBudget} label="New budget" />
         }
       />
-      <Accounts
-        accounts={accounts}
+      <Budgets
+        budgets={budgets}
         onNewAllocation={onNewAllocation}
+        onEdit={onEdit}
+        onDeactivate={onDeactivate}
+        onReactivate={onReactivate}
+      />
+      <AllocationsHistory allocations={allocations} />
+      <DeactivateModal
+        state={modal}
+        onClose={closeModal}
+        onSubmit={onSubmitDeactivate}
       />
     </Wrap>
   )
 }
 
-// eslint-disable-next-line import/no-unused-modules
+const DeactivateModal = ({ state, onClose, onSubmit }) => {
```

Comment:
I think this could definitely be moved to an external component to avoid overloading the main App component